clinica/iotools/converters/ixi_to_bids/ixi_to_bids_utils.py

```diff
@@ -0,0 +1,436 @@
+        {
+            "filename": [
+                f"{path.parent.name}/{path.name}"
+                for path in (bids_dir / bids_id).rglob(f"{bids_id}*.nii.gz")
+            ]
+        }
+    )
+    to_write.to_csv(
+        bids_dir / bids_id / "ses-M000" / f"{bids_id}_ses-M000_scans.tsv",
+        sep="\t",
+        index=False,
+    )
+
+
+def write_sessions(
+    bids_dir: Path, clinical_data: pd.DataFrame, participant: str
+) -> None:
+    """
+    Writes the sessions.tsv for a IXI subject.
+
+    Parameters
+    ----------
+    bids_dir : Path to the output BIDS directory.
+    clinical_data : Dataframe containing the formatted clinical data of the IXI study.
+    participant : Current converted subject study id (str).
+    """
+    line = clinical_data[clinical_data["source_id"] == participant]
+    bids_id = bids_id_factory(StudyName.IXI).from_original_study_id(participant)
+    line[["source_id", "session_id", "acq_time"]].to_csv(
+        bids_dir / bids_id / f"{bids_id}_sessions.tsv", sep="\t", index=False
+    )
+
+
+def write_participants(
+    bids_dir: Path, clinical_data: pd.DataFrame, participants: List[str]
+) -> None:
+    """
+    Write the participants.tsv at the root of the BIDS directory.
+
+    Parameters
+    ----------
+    bids_dir : Path to the output BIDS directory.
+    clinical_data : Dataframe containing the formatted clinical data of the IXI study.
+    participants : List of converted subjects study source ids.
+    """
+    clinical_data.set_index("source_id", inplace=True, drop=False)
+    clinical_data.assign(
+        participant_id=clinical_data.source_id.apply(
+            lambda x: bids_id_factory(StudyName.IXI).from_original_study_id(x)
+        )
+    )
+    for participant in participants:
+        if participant not in clinical_data.index:
+            clinical_data.loc[participant] = "n/a"
+            clinical_data.loc[participant, "source_id"] = participant
+    if not bids_dir.exists():
+        bids_dir.mkdir()
+    clinical_data.loc[participants].drop(["acq_time", "session_id"], axis=1).to_csv(
+        bids_dir / "participants.tsv", sep="\t", index=False, na_rep="n/a"
+    )
+    clinical_data.reset_index(drop=True, inplace=True)
+
+
+def _identify_expected_modalities(data_directory: Path) -> List[str]:
+    return [
+        p.name.split("-")[1]
+        for p in data_directory.iterdir()
+        if p.is_dir() and "IXI-" in str(p)
+    ]
+
+
+def check_modalities(data_directory: Path, participants: List[str]) -> None:
+    """
+    Verify what modality folders are available in the given data directory and checks if some are missing per participant
+
+    Parameters
+    ----------
+    data_directory : Path to raw dataset.
+    participants : List of the subject ids of all participants
+
+    """
+    expected_modalities = _identify_expected_modalities(data_directory)
+    message = (
+        f"Modalities : {' , '.join(_rename_modalities(mod) for mod in expected_modalities)}"
```

Comment:
The unit tests are failing because this message is not deterministic: it depends on the order of `expected_modalities`, which in turn depends on the folder iteration logic in `_identify_expected_modalities`. Depending on the platform, this can vary.
A simple and quick solution is to sort the `expected_modalities` list before displaying the message.